### PR TITLE
Add reduction-effects to the CI

### PR DESCRIPTION
**Pull request by JasonGross**
**Kind:** infrastructure

### CODE REVIEW

**Comment by JasonGross** [comment]
I'm confused by the CI failure:
```
+ tar xvfz b8dd8290f0648751cbf64456435cc284144cb0d2.tar.gz --strip-components=1
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/.circleci/
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/.circleci/config.yml
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/.gitignore
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/.gitmodules
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/CHANGELOG.md
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/LICENSE
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/Makefile
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/PrintingEffect.v
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/README.md
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/_CoqProject
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/coq-reduction-effects.opam
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/meta.yml
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/redeffect.mlg
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/redeffect_plugin.mlpack
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/templates/
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/tests/
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/tests/Makefile
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/tests/Makefile.coq.local
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/tests/PrintEffect.expected
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/tests/PrintEffect.v
 reduction-effects-b8dd8290f0648751cbf64456435cc284144cb0d2/tests/_CoqProject
 + rm -f b8dd8290f0648751cbf64456435cc284144cb0d2.tar.gz
 + cd /builds/coq/coq/_build_ci/reduction-effects
 /builds/coq/coq/dev/ci/ci-reduction-effects.sh: line 8: cd: /builds/coq/coq/_build_ci/reduction-effects: No such file or directory
 Makefile.ci:75: recipe for target 'ci-reduction-effects' failed
```
Did I do something wrong?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -749,6 +749,9 @@ plugin:plugin-tutorial:
 plugin:ci-quickchick:
   extends: .ci-template-flambda
 
+plugin:ci-reduction-effects:
```
Comment (SkySkimmer) [inline .gitlab-ci.yml]:
```suggestion
plugin:ci-reduction_effects:
```

File: Makefile.ci [hunk 2]
```diff
@@ -38,6 +38,7 @@ CI_TARGETS= \
     ci-paramcoq \
     ci-perennial \
     ci-quickchick \
+    ci-reduction-effects \
```
Comment (SkySkimmer) [inline Makefile.ci]:
```suggestion
    ci-reduction_effects \
```

File: dev/ci/ci-reduction-effects.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+ci_dir="$(dirname "$0")"
+. "${ci_dir}/ci-common.sh"
+
+git_download reduction_effects
+
+( cd "${CI_BUILD_DIR}/reduction-effects" && make && make test && make install)
```
Comment (SkySkimmer) [inline dev/ci/ci-reduction-effects.sh]:
```suggestion
( cd "${CI_BUILD_DIR}/reduction_effects" && make && make test && make install)
```

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
See comments, and also change the .sh name to `ci-reduction_effects.sh`

**Comment by JasonGross** [comment]
@SkySkimmer Thanks, should be fixed now

**Comment by JasonGross** [comment]
CI passes now

**Review by SkySkimmer — APPROVED** [review]

**Comment by ppedrot** [comment]
Is this really supposed to go into 8.11?

**Comment by JasonGross** [comment]
I don't see why it shouldn't go into 8.11 (and it would be nice to have a version of it distributed with the Windows Coq installer, if that's on the table), but if you think it's a bad idea or not work it, feel free to drop it from 8.11.

**Comment by ppedrot** [comment]
Merely backporting this PR will not add it to the Windows installer, and furthermore this will require clever pinning of the dev in 8.11. So, I don't think it is worth it.

**Pull request merged**